tests/utils/deployment/deployment.go

```diff
@@ -81,3 +96,42 @@ func RedefineWithPreStopSpec(deployment *v1.Deployment, command []string) *v1.De
 
 	return deployment
 }
+
+func RedefineFirstContainerWithPreStopSpec(deployment *v1.Deployment, command []string) *v1.Deployment {
+	deployment.Spec.Template.Spec.Containers[0].Lifecycle = &corev1.Lifecycle{
```

Comment:
You should assert that "len(Containers) > 0" before accessing its first element.